### ISSUE TITLE
curl commands are not rendered with security headers

### ISSUE THREAD

**Issue by hotgazpacho** [wdullaer/raml2html-slate-theme#28]
When specifying that a resource is `securedBy` as entry in the `securitySchemes`, the curl command generated for that resource does not include the relevant headers, thus creating an incomplete curl request. The supported security schemes list is finite, and most involve setting HTTP headers, such as `Authorization` or `X-API-Key`. The definitions of securitySchemes are pretty expressive, too.

**Comment by hotgazpacho** [comment]
I've started work on addressing this in a branch on my fork: https://github.com/wdullaer/raml2html-slate-theme/compare/master...hotgazpacho:curl-security-schemes?expand=1

I've got a very simple test case for basic auth working, with one entry in the `securitySchemes`, and a matching entry in `securedBy`. It is ugly, but functional. Clearly a first stab, so I'll eventually refactor. One of the things that I'm not happy with is appending the security schemas parameter at the end of the function's signature. This should probably become the first parameter, and bind it in the `renderHtml` function on `index.js`, like is done with the `baseUri`. However, I thought changing all the `getCurlStatement` tests in the first go-round was too big a step.
